### PR TITLE
feat: add cilium e2e tests

### PR DESCRIPTION
**Pull request by frezbo**
Add cilium e2e tests. The existing cilium check was very old, update to latest cilium version and also add a test for KPR strict mode.

### CODE REVIEW

File: .drone.jsonnet [hunk 1]
```diff
@@ -420,10 +420,17 @@ local integration_extensions = Step('e2e-extensions', target='e2e-qemu', privile
   WITH_TEST: 'run_extensions_test',
   IMAGE_REGISTRY: local_registry,
 });
-local integration_cilium = Step('e2e-cilium-1.9.10', target='e2e-qemu', privileged=true, depends_on=[integration_extensions], environment={
+local integration_cilium = Step('e2e-cilium', target='e2e-qemu', privileged=true, depends_on=[integration_extensions], environment={
```
Comment (smira) [inline .drone.jsonnet:423]: I feel like we should probably split `integration-misc` a bit, as it now might run for 1.5 hours with the `extensions` build... nothing pressing, probably `e2e-extensions` should be its own pipeline, as it got significantly more complex?

Comment (frezbo) [reply .drone.jsonnet:423]: yes, makes sense, should cilium also be it's own pipeline, also would make my testing easier :sweat_smile:

File: hack/test/e2e-qemu.sh [hunk 2]
```diff
@@ -129,6 +129,16 @@ case "${WITH_CONFIG_PATCH_FILE_WORKER:-false}" in
     ;;
 esac
 
+case "${WITH_SKIP_BOOT_PHASE_FINISHED_CHECK:-no}" in
+  # using arrays here to preserve spaces properly in WITH_CONFIG_PATCH_FILE
```
Comment (smira) [inline hack/test/e2e-qemu.sh:133]: comment here doesn't match?

**Review by smira — APPROVED** [review]

File: .drone.jsonnet (hunk 1 above)
Comment (smira) [reply .drone.jsonnet:423]: as I was testing this, I commented out everything but Cilium from the `integration-misc`, or make it the first one - the order doesn't matter

Comment (frezbo) [reply .drone.jsonnet:423]: > as it now might run for 1.5 hours with the extensions build

extensions e2e is a different pipeline, it's only part of intergration-misc as part of cron, should we change it to a separate cron?

Comment (smira) [reply .drone.jsonnet:423]: yes, probably

**Pull request merged**